langchain4j-langfuse/src/main/java/dev/langchain4j/langfuse/LangfuseClient.java

```diff
@@ -0,0 +1,216 @@
+                    .body(MAPPER.writeValueAsString(Map.of(
+                            "output", output,
+                            "status", status)))
+                    .build();
+
+            SuccessfulHttpResponse response = httpClient.execute(request);
+            if (response.statusCode() >= 300) {
+                throw new IOException("Failed to update observation: " + response.body());
+            }
+        }
+    }
+
+    public class SessionAPI {
+        public void create(Session session) throws IOException {
+            HttpRequest request = baseRequestBuilder()
+                    .method(HttpMethod.POST)
+                    .url(endpoint + "/api/public/sessions")
+                    .body(MAPPER.writeValueAsString(session))
+                    .build();
+
+            SuccessfulHttpResponse response = httpClient.execute(request);
+            if (response.statusCode() >= 300) {
+                throw new IOException("Failed to create session: " + response.body());
+            }
+        }
+
+        public void addTrace(String sessionId, String traceId) throws IOException {
+            HttpRequest request = baseRequestBuilder()
+                    .method(HttpMethod.POST)
+                    .url(endpoint + "/api/public/sessions/" + sessionId + "/traces")
+                    .body(MAPPER.writeValueAsString(Map.of("traceId", traceId)))
+                    .build();
+
+            SuccessfulHttpResponse response = httpClient.execute(request);
+            if (response.statusCode() >= 300) {
+                throw new IOException("Failed to add trace to session: " + response.body());
+            }
+        }
+    }
+
+    public class ScoreAPI {
+        public void create(Score score) throws IOException {
+            HttpRequest request = baseRequestBuilder()
+                    .method(HttpMethod.POST)
+                    .url(endpoint + "/api/public/scores")
+                    .body(MAPPER.writeValueAsString(score))
+                    .build();
+
+            SuccessfulHttpResponse response = httpClient.execute(request);
+            if (response.statusCode() >= 300) {
+                throw new IOException("Failed to create score: " + response.body());
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        if (httpClient instanceof AutoCloseable) {
+            try {
+                ((AutoCloseable) httpClient).close();
+            } catch (Exception e) {
+                log.warn("Error closing HTTP client", e);
+            }
+        }
+    }
+
+    public void shutdown() {
+        close();
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private String endpoint = "https://cloud.langfuse.com";
+        private String publicKey;
+        private String secretKey;
+        private HttpClient httpClient;
+
+        public Builder endpoint(String endpoint) {
```

Comment:
```suggestion
        public Builder baseUrl(String baseUrl) {
```
Please use `baseUrl` name instead of `endpoint` (everywhere)